build_scripts/build_usd.py

```diff
@@ -2043,14 +2108,18 @@ def _JoinVersion(v):
         sys.exit(1)
 
 if PYSIDE in requiredDependencies:
+    # Special case - we are given the PYSIDEUICBINARY as cmake arg.
+    usdBuildArgs = context.GetBuildArguments(USD)
+    given_pysideUic = 'PYSIDEUICBINARY' in " ".join(usdBuildArgs)
```

Comment:
pyside-uic has been deprecated by Qt. So added ability to pass in what to use.

Reply:
Is pyside-uic deprecated? I thought it was just getting complicated to distribute with py2 vs py3 and pyside1 vs pyside2. 

The actual source of pyside-uic/pyside2-uic/etc is pretty small. The latest version I have just imports an object called `uic` and executes it. Older versions in pyside and pyside2 were pretty minimal as well, though not as short. I wonder if we could include our own pyside-uic with USD's build scripts and run it instead of making a user install extra packages and find their executable name. If we did that we could also get rid of all the name checking we do below this comment line.

This issue is related https://github.com/PixarAnimationStudios/USD/issues/1419

Reply:
Yes as Hamed mentions below uic is now capable of generating python code. Please see my change in "cmake/macros/Private.cmake" where I add the python generator if using uic. There is no more pyside2-uic.

Reply:
Since pyside2-uic appears to be deprecated (although the latest PyPI package still seems to include it?), it seems like the build should recognize `uic` without it having to be passed in via a special build flag. So how about just adding `uic` to the list of programs in `pyside2Uic` instead of checking PYSIDEUICBINARY?

Reply:
Yes we could add it to the list of executables to search for, but having a way to pass in a pyside executable to use is very helpful. This code assumes the one you want to use is in the path.

Reply:
Added 'uic' to list of pyside executables.